nilearn/decoding/decoder.py

```diff
@@ -793,7 +775,7 @@ def fit(self, X, y, groups=None):
                 classes_, self.coef_, self.std_coef_
             )
 
-            if self._is_classification and (self.n_classes_ == 2):
+            if is_classifier(self) and (self.n_classes_ == 2):
```

Comment:
Could also the part between lines 754-794 be exracted to an abstract method and be implemented in the concrete class? 

At the same time, I did not understand how is it possible for estimator to be a DummyClassifier or a DummyRegressor. May be what I propose is not convenient.